relatio/_embeddings.py

```diff
@@ -98,29 +101,39 @@ def use_sif(self) -> bool:
         return self._use_sif
 
     # One cannot add a setter since it is added next to the child classes
-
     def get_vector(self, phrase: str) -> Optional[np.ndarray]:
         tokens = phrase.split()
 
-        # For phrases with one token
-        if self.use_sif is True and len(tokens) > 0:
-
+        if self.use_sif:
+
+            sif_tokens = []
+            for token in tokens:
+                if token in self._sif_dict:
+                    sif_tokens.append(token)
+                else:
+                    warnings.warn(
+                        f"No frequency information for token: {token}. It is not used to compute the SIF-embedding of phrase: {phrase}.",
+                        RuntimeWarning,
+                    )
             res = np.mean(
                 [
                     self._sif_dict[token] * self._get_default_vector(token)
-                    for token in tokens
+                    for token in sif_tokens
                 ],
                 axis=0,
             )
-
         else:
             res = self._get_default_vector(phrase)
 
-        # in case the result is fishy it will return a None
-        if res is None:
-            return None
-        elif np.isnan(res).any() or np.count_nonzero(res) == 0:
-            return None
+        # In case the result is fishy it will return a vector of np.nans and raise a warning
+        if res is None or np.isnan(res).any() or np.count_nonzero(res) == 0:
```

Comment:
Can we still get a `None`?
If yess I like to keep it as a different option in the if statement because it is a different type that is not compatible with`np.isnan(res).any() or np.count_nonzero(res)`.

Reply:
I see. I don't think we still can, indeed

Reply:
remove None